end_to_end/end_to_end_suite_test.go

```diff
@@ -2263,6 +2263,21 @@ LANGUAGE plpgsql NO SQL;`)
 			assertArtifactsCleaned(restoreConn, "20240502095933")
 			testhelper.AssertQueryRuns(restoreConn, "DROP TABLE t0; DROP TABLE t1; DROP TABLE t2; DROP TABLE t3; DROP TABLE t4;")
 		})
+		It("Will not hang after error during restore with jobs", func() {
+			command := exec.Command("tar", "-xzf", "resources/2-segment-db-error.tar.gz", "-C", backupDir)
```

Comment:
This test can't be run on demo cluster:

```
      20240506:14:39:43 gprestore:givi:samsung-ivanov:120268-[INFO]:-Restoring pre-data metadata
Pre-data objects restored:  0 / 14 [-------------------------------------------------------]   0.00%20240506:14:39:43 gprestore:givi:samsung-ivanov:120268-[CRITICAL]:-ERROR: role "gpadmin" does not exist (SQLSTATE 42704): Error encountered while creating schema public
```

Reply:
I run this test on demo cluster, moreover data for this test is taken from another merged patch

Reply:
this test should be reworked later